### PR TITLE
Dtspb 3438 superuser change case state

### PR DESCRIPTION
**Pull request by MarcusHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-3438

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -860,5 +860,6 @@
   {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "deathRecordVerificationFailed", "UserRole": "citizen", "CRUD": "CRUD"},
   {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "makeCaseDormant", "UserRole": "citizen", "CRUD": "R"},
   {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "reactivateDormantCase", "UserRole": "citizen", "CRUD": "R"},
-  {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "registrarsDecision", "UserRole": "citizen", "CRUD": "R"}
+  {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "registrarsDecision", "UserRole": "citizen", "CRUD": "R"},
+  {"CaseTypeID": "GrantOfRepresentation", "CaseEventID": "changeState", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"}
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseEvent.json]: can you add all the superuser auths into the end of the existing block for them please?

File: ccdImports/configFiles/CCD_Probate_Backoffice/FixedLists.json [hunk 2]
```diff
@@ -354,5 +354,25 @@
   {"LiveFrom": "02/01/2017", "ID": "registrarsDecisionType", "ListElementCode": "probateRefused", "ListElement": "Probate refused", "DisplayOrder": 1},
   {"LiveFrom": "02/01/2017", "ID": "registrarsDecisionType", "ListElementCode": "lostWill", "ListElement": "Rule 54 (lost will)", "DisplayOrder": 2},
   {"LiveFrom": "02/01/2017", "ID": "registrarsDecisionType", "ListElementCode": "otherOrder", "ListElement": "Other order / direction", "DisplayOrder": 3},
-  {"LiveFrom": "02/01/2017", "ID": "registrarsDecisionType", "ListElementCode": "refusal", "ListElement": "Refusal", "DisplayOrder": 4}
+  {"LiveFrom": "02/01/2017", "ID": "registrarsDecisionType", "ListElementCode": "refusal", "ListElement": "Refusal", "DisplayOrder": 4},
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Backoffice/FixedLists.json:357]: probably want to put a display order on these - alphabetical?

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java [hunk 3]
```diff
@@ -145,7 +146,8 @@ public CallbackResponse updateTaskList(CallbackRequest callbackRequest) {
     }
 
     public CallbackResponse defaultDateOfDeathType(CallbackRequest callbackRequest) {
-        ResponseCaseDataBuilder<?, ?> builder = ResponseCaseData.builder().dateOfDeathType(DEFAULT_DATE_OF_DEATHTYPE);
+        ResponseCaseDataBuilder<?, ?> builder =
+                ResponseCaseData.builder().dateOfDeathType(DEFAULT_DATE_OF_DEATHTYPE).state("HelloKitty");
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: whats this? Hello Kitty?

File: ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseField.json [hunk 4]
```diff
@@ -2128,6 +2128,8 @@
   {"CaseTypeID": "GrantOfRepresentation", "CaseFieldID": "lastEvidenceAddedDate", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"},
   {"CaseTypeID": "GrantOfRepresentation", "CaseFieldID": "registrarDirections", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"},
   {"CaseTypeID": "GrantOfRepresentation", "CaseFieldID": "registrarDirectionToAdd", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"},
+  {"CaseTypeID": "GrantOfRepresentation", "CaseFieldID": "transferToState", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"},
+  {"CaseTypeID": "GrantOfRepresentation", "CaseFieldID": "currentApplicationStateName", "UserRole": "caseworker-probate-superuser", "CRUD": "CRUD"},
```
Comment (sanjayparekh-hmcts) [inline ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseField.json]: I believe you can use a meta data field to tell the user the current state - we dont need to save a new field
contact me!

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java [hunk 5]
```diff
@@ -2344,14 +2364,229 @@ void shouldResolveStopCaseReadyToIssue() {
...
+                .resolveStopState(CASE_MATCHING_ISSUE_GRANT);
 
         when(callbackRequestMock.getCaseDetails()).thenReturn(caseDetailsMock);
         when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
         CallbackResponse callbackResponse = underTest.resolveStop(callbackRequestMock);
         assertEquals(CASE_MATCHING_ISSUE_GRANT, callbackResponse.getData().getState());
     }
 
+    @Test
+    void shouldChangeCaseStateBOCaseQA() {
```
Comment (sanjayparekh-hmcts) [inline src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java:2358]: maybe reduce these tests as 'duplicates'

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java [hunk 6]
```diff
@@ -52,6 +52,7 @@ public class ResponseCaseData extends ResponseCaseDataParent {
     private final String state;
     private final ApplicationType applicationType;
     private final String registryLocation;
+    private final String currentApplicationStateName;
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java]: I dont think you need this any more?

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java (hunk 3 above)
Comment (MarcusHMCTS) [reply src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: deleted

File: ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseField.json (hunk 4 above)
Comment (MarcusHMCTS) [reply ccdImports/configFiles/CCD_Probate_Backoffice/AuthorisationCaseField.json]: use CCD MetaDate instead

File: src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java (hunk 5 above)
Comment (MarcusHMCTS) [reply src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java:2358]: removed

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java (hunk 6 above)
Comment (MarcusHMCTS) [reply src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseData.java]: removed

**Pull request closed without merging**